src/Polly.Core/ResilienceStrategy.Sync.cs

```diff
@@ -31,6 +31,63 @@ public abstract partial class ResilienceStrategy
             (callback, state)).GetResult();
     }
 
+    /// <summary>
+    /// Executes the specified callback.
+    /// </summary>
+    /// <param name="callback">The user-provided callback.</param>
+    /// <param name="context">The context associated with the callback.</param>
+    public void Execute(
+        Action<ResilienceContext> callback,
+        ResilienceContext context)
+    {
+        Guard.NotNull(callback);
+        Guard.NotNull(context);
+
+        InitializeSyncContext(context);
+
+        ExecuteCoreAsync(
+            static (context, state) =>
+            {
+                state(context);
+                return new ValueTask<VoidResult>(VoidResult.Instance);
```

Comment:
Is there any value in having this cached as a static somewhere, or is that pointless because it's a `ValueTask<T>` struct?

Reply:
If this was a class I would say yes, but with the structs I don't think it will make a difference.